web/Controllers/AdminController.cs

```diff
@@ -133,20 +136,37 @@ public async Task<ActionResult> GetAdminList()
 
     [Authorize(Roles = "Admin")]
     [HttpGet]
-    public async Task<IActionResult> LandlordList(LandlordListModel landlordListModel)
+    public IActionResult LandlordList(bool? isApproved = null, bool? isAssigned = null, int page = 1, int landlordsPerPage = 10)
     {
-        landlordListModel.LandlordList = await _adminService.GetLandlordList(landlordListModel);
-        return View(landlordListModel);
+        var landlords = _adminService.GetLandlordList(isApproved, isAssigned);

```

Comment:
so I would definitely suggest adding in the pagination into the service, and then calling that here. That way we can be certain the pagination is called before any ToList, and that the pagination happens in the DB rather than in memory :)

Reply:
In order for it to work with how this is working, you could have the method take in the pagination parameters as well and then return both the list and the count